### PR TITLE
Add some dataplane utilities

### PR DESCRIPTION
**Pull request by DanG100**

### CODE REVIEW

File: dataplane/internal/engine/table.go [hunk 1]
```diff
@@ -0,0 +1,300 @@
...
+	if _, err := c.TableCreate(ctx, neighbor); err != nil {
+		return err
+	}
+	if err := createFIBSelector(ctx, c); err != nil {
+		return err
+	}
+	return nil
+}
+
+func createFIBSelector(ctx context.Context, c fwdpb.ServiceClient) error {
```
Comment (wenovus) [inline dataplane/internal/engine/table.go:130]: A short comment please.

File: dataplane/internal/engine/table.go [hunk 2]
```diff
@@ -0,0 +1,300 @@
...
+func mustParseHex(hexStr string) []byte {
+	b, err := hex.DecodeString(hexStr)
+	if err != nil {
+		panic(err)
+	}
+	return b
+}
+
+// SetupTables creates the tables used for routing.
+func SetupTables(ctx context.Context, c fwdpb.ServiceClient) error {
```
Comment (wenovus) [inline dataplane/internal/engine/table.go:36]: "tables used for routing" -> routing tables? or forwarding tables?

File: dataplane/internal/engine/port.go [hunk 3]
```diff
@@ -0,0 +1,265 @@
...
+			},
+		},
+	}
+	if _, err := c.PortUpdate(ctx, update); err != nil {
+		return err
+	}
+	return nil
+}
+
+func puntEtherTypeTable(port string) string {
```
Comment (wenovus) [inline dataplane/internal/engine/port.go:118]: I'm not quite understanding why this table is called "punt" and "ethertype", could you explain a little?

File: dataplane/internal/engine/port.go [hunk 4]
```diff
@@ -0,0 +1,282 @@
...
+		},
+	}
+	if _, err := c.PortUpdate(ctx, update); err != nil {
+		return err
+	}
+	return nil
+}
+
+// puntEtherTypeTable returns the name of the table containing the punt rules based on the EtherType packet header field.
+func puntEtherTypeTable(port string) string {
```
Comment (wenovus) [inline dataplane/internal/engine/port.go:134]: etherTypePuntTable -- I think this is clearer

File: dataplane/internal/engine/port.go [hunk 5]
```diff
@@ -0,0 +1,282 @@
...
+	return fmt.Sprintf("%s-punt-etherType", port)
+}
+
+// puntEtherTypeTable returns the name of the table containing the punt rules based on the IP protocol packet header field.
+func puntIPProtoType(port string) string {
```
Comment (wenovus) [inline dataplane/internal/engine/port.go:139]: IPProtocolNumPuntTable, also update comment.

File: dataplane/internal/engine/port.go [hunk 6]
```diff
@@ -0,0 +1,282 @@
...
+	return fmt.Sprintf("%s-punt-ipproto", port)
+}
+
+func setupPuntRules(ctx context.Context, c fwdpb.ServiceClient, name string) error {
```
Comment (wenovus) [inline dataplane/internal/engine/port.go:143]: so the name is a port name?

Comment (wenovus) [inline dataplane/internal/engine/port.go:143]: setupPuntTables to use the same "table" word as SetupTables? Maybe rename that one to SetupForwardingTables?

File: dataplane/internal/engine/port.go [hunk 7]
```diff
@@ -0,0 +1,282 @@
...
+
+// CreateLocalPort creates an local (ie TAP) port.
+func CreateLocalPort(ctx context.Context, c fwdpb.ServiceClient, name string) error {
+	port := &fwdpb.PortCreateRequest{
+		ContextId: &fwdpb.ContextId{Id: contextID},
+		Port: &fwdpb.PortDesc{
+			PortType: fwdpb.PortType_PORT_TYPE_KERNEL,
+			Port: &fwdpb.PortDesc_Kernel{
+				Kernel: &fwdpb.KernelPortDesc{
+					DeviceName: name,
```
Comment (wenovus) [inline dataplane/internal/engine/port.go:100]: does this `name` need to be converted to a TAP name or no? Either way should this be documented in the docstring?

**Review by wenovus — APPROVED** [review]
Thanks for adding the comments, was much easier to review.

File: dataplane/internal/engine/port.go (hunk 6 above)
Comment (DanG100) [reply dataplane/internal/engine/port.go]: yes

**Pull request merged**